tests/test_app_cli.py

```diff
@@ -173,11 +173,14 @@ async def test_export(tmpvenv_active_inherit: env.VirtualEnv, tmpdir, server, cl
 """
     )
 
+    # Use non-default agent to make sure the resource can be checked in the 'deploying' state.
+    # Using the internal default agent might trigger a race condition where the 'success' state
+    # is reached because the resource actually gets deployed.
+
     path_main_file.write(
         """
 import std::testing
-vm1=std::Host(name="non-existing-machine", os=std::linux)
-std::testing::NullResource(name=vm1.name)
+std::testing::NullResource(name="test", agentname="not_internal")
```

Comment:
This makes more clear what the intention is.
```suggestion
std::testing::NullResource(name="test", agentname="non_existing_agent")
```